src/compile/common.ts

```diff
@@ -210,26 +213,25 @@ export function mergeTitleFieldDefs(f1: readonly FieldDefBase<string>[], f2: rea
   return merged;
 }
 
-export function mergeTitle(title1: string, title2: string) {
-  if (title1 === title2 || !title2) {
+export function mergeTitle(title1: Text, title2: Text) {
+  if (deepEqual(title1, title2) || !title2) {
     // if titles are the same or title2 is falsy
     return title1;
   } else if (!title1) {
     // if title1 is falsy
     return title2;
   } else {
-    // join title with comma if they are different
-    return title1 + ', ' + title2;
+    return [...array(title1), ...array(title2)].join(', ');
   }
 }
 
 export function mergeTitleComponent(v1: Explicit<AxisTitleComponent>, v2: Explicit<AxisTitleComponent>) {
-  if (isArray(v1.value) && isArray(v2.value)) {
+  if (!isText(v1.value) && !isText(v2.value)) {
     return {
       explicit: v1.explicit,
       value: mergeTitleFieldDefs(v1.value, v2.value)
     };
-  } else if (!isArray(v1.value) && !isArray(v2.value)) {
+  } else if (isText(v1.value) && isText(v2.value)) {
     return {
       explicit: v1.explicit, // keep the old explicit
       value: mergeTitle(v1.value, v2.value)
```

Comment:
Is it not possible to merge `null` and "title"? If so, we need an else case?

Reply:
Good catch. Fixed